testing/dart/codec_test.dart

```diff
@@ -41,6 +41,35 @@ void main() {
     ui.Codec codec = await completer.future;
     expect(codec, null);
   });
+
+  test('nextFrame fails when no callback provided', () async {
+    Uint8List data = await _getSkiaResource('alphabetAnim.gif').readAsBytes();
+    Completer<ui.Codec> completer = new Completer<ui.Codec>();
+    expect(ui.instantiateImageCodec(data, completer.complete), null);
+    ui.Codec codec = await completer.future;
+    expect(codec.getNextFrame(null), 'Callback must be a function');
+  });
+
+  test('nextFrame', () async {
+    Uint8List data = await _getSkiaResource('test640x479.gif').readAsBytes();
+    Completer<ui.Codec> completer = new Completer<ui.Codec>();
+    expect(ui.instantiateImageCodec(data, completer.complete), null);
+    ui.Codec codec = await completer.future;
+    List<String> decodedFrameInfos = [];
+    for (int i = 0; i < 5; i++) {
+      Completer<ui.FrameInfo> frameCompleter = new Completer<ui.FrameInfo>();
+      codec.getNextFrame(frameCompleter.complete);
+      ui.FrameInfo frameInfo = await frameCompleter.future;
+      decodedFrameInfos.add("${frameInfo.durationMillis} ${frameInfo.image}");
+    }
+    expect(decodedFrameInfos, equals([
+      "200 [640\u00D7479]",
```

Comment:
It would be clearer to compare against a list of [duration, image.width, image.height] lists instead of relying on the implementation of toString()

Reply:
done